core/tests/unit/test__http.py

```diff
@@ -119,129 +139,123 @@ def test_build_api_url_w_extra_query_params(self):
-        self.assertEqual(http._called_with['uri'], URI)
-        self.assertIsNone(http._called_with['body'])
+        path = '/path/required'
+
+        self.assertEqual(conn.api_request('GET', path), {})
+
         expected_headers = {
             'Accept-Encoding': 'gzip',
-            'Content-Length': '0',
             'User-Agent': conn.USER_AGENT,
         }
-        self.assertEqual(http._called_with['headers'], expected_headers)
+        expected_url = '{base}/mock/{version}{path}'.format(
+            base=conn.API_BASE_URL,
+            version=conn.API_VERSION,
+            path=path)
+        http.request.assert_called_once_with(
+            method='GET',
+            url=expected_url,
+            headers=expected_headers,
+            data=None)
 
     def test_api_request_w_non_json_response(self):
-        http = _Http(
-            {'status': '200', 'content-type': 'text/plain'},
-            b'CONTENT',
-        )
+        http = make_requests_session([
+            make_response(content=b'content')])
         client = mock.Mock(_http=http, spec=['_http'])
         conn = self._make_mock_one(client)
 
-        self.assertRaises(TypeError, conn.api_request, 'GET', '/')
+        with self.assertRaises(ValueError):
+            conn.api_request('GET', '/')
 
     def test_api_request_wo_json_expected(self):
-        http = _Http(
-            {'status': '200', 'content-type': 'text/plain'},
-            b'CONTENT',
-        )
+        http = make_requests_session([
+            make_response(content=b'content')])
         client = mock.Mock(_http=http, spec=['_http'])
         conn = self._make_mock_one(client)
-        self.assertEqual(conn.api_request('GET', '/', expect_json=False),
-                         b'CONTENT')
+
+        result = conn.api_request('GET', '/', expect_json=False)
+
+        self.assertEqual(result, b'content')
 
     def test_api_request_w_query_params(self):
         from six.moves.urllib.parse import parse_qs
         from six.moves.urllib.parse import urlsplit
 
-        http = _Http(
-            {'status': '200', 'content-type': 'application/json'},
-            b'{}',
-        )
+        http = make_requests_session([self.EMPTY_JSON_RESPONSE])
         client = mock.Mock(_http=http, spec=['_http'])
         conn = self._make_mock_one(client)
-        self.assertEqual(conn.api_request('GET', '/', {
+
+        result = conn.api_request('GET', '/', {
             'foo': 'bar',
             'baz': ['qux', 'quux']
-        }), {})
-        self.assertEqual(http._called_with['method'], 'GET')
-        uri = http._called_with['uri']
-        scheme, netloc, path, qs, _ = urlsplit(uri)
+        })
+
+        self.assertEqual(result, {})
+
+        expected_headers = {
+            'Accept-Encoding': 'gzip',
+            'User-Agent': conn.USER_AGENT,
+        }
+        http.request.assert_called_once_with(
+            method='GET', url=mock.ANY, headers=expected_headers,
```

Comment:
The URL is non-trivial, the actual contents of the url and its query string are checked in the next section.